streams/src/main/java/org/apache/kafka/streams/state/internals/KeyValueToTimestampedKeyValueByteStoreAdapter.java

```diff
@@ -180,5 +201,38 @@ public <PS extends Serializer<P>, P> KeyValueIterator<Bytes, byte[]> prefixScan(
     public long approximateNumEntries() {
         return store.approximateNumEntries();
     }
+}
 
+class WrappedRocksDbIterator extends AbstractIterator implements ManagedKeyValueIterator {
+    final private RocksDbIterator rocksDbIterator;
+    public WrappedRocksDbIterator(final RocksDbIterator rocksDbIterator) {
+        this.rocksDbIterator = rocksDbIterator;
+    }
+    @Override
+    protected Object makeNext() {
+        final KeyValue<Bytes, byte[]> next = rocksDbIterator.makeNext();
```

Comment:
Given that we _wrap_ `RocksDbIterator` we should not call `makeNext` directly? But instead of public methods only, ie, is `hasNext()` and `next()` instead.

Reply:
Not sure if I understand what you are trying to say? Can you elaborate?

`makeNext` is not public API, but `protected` -- because this code "extends" `AbstractIterator` we are allowed to call `makeNext` but it seem not to be clean... Cf https://github.com/apache/kafka/pull/14570#discussion_r1379541488 -- If we change this class to only `implements MangedKeyValueIterator` I believe we won't be able to call `makeNext` any longer, what indicates that we should not do it.

Reply:
In the `RockDbIterator`,` makeNext` is public, so I change our method from protected to public